src/vs/workbench/services/editor/common/editorOverrideService.ts

```diff
@@ -260,7 +260,7 @@ export function globMatchesResource(globPattern: string | glob.IRelativePattern,
 		return false;
 	}
 	const matchOnPath = typeof globPattern === 'string' && globPattern.indexOf(posix.sep) >= 0;
-	const target = matchOnPath ? resource.path : basename(resource);
+	const target = matchOnPath ? `${resource.scheme}:${resource.path}` : basename(resource);
```

Comment:
I'm suspicious of including the scheme in a glob, that isn't typical. Where do the glob patterns come from? This could make my `foo` glob stop matching `/foo`

Reply:
This doesn't change glob matching logic, just for custom editors and notebooks. The glob patterns come from all the registers, normally in the from of `*.ext`, but can be anything. For notebooks and custom editors they primarily come from extensions

Reply:
Yeah but are extensions writing globs like `vscode-notebook-cell://**/...`? Or where does that come from?

Reply:
The extensions write globs likes `*.md` and then I saw that since they're a notebook they also handle cell URIs so I form the glob `vscode-notebook-cell:/**/*.md`

Reply:
I guess it's ok for a candidate, but I think a better way to do this would be to register the scheme and file pattern separately, like `{ scheme: 'vscode-notebook-cell', pattern: extensionGlobPattern }`. These are separate things and it feels weird to combine them